rcl_yaml_param_parser/src/parse.c

```diff
@@ -786,13 +786,6 @@ rcutils_ret_t parse_key(
       {
         /// Till we get PARAMS_KEY or PARAMS_DESCRIPTORS_KEY, keep adding to node namespace
         if (0 == strncmp(PARAMS_KEY, value, strlen(PARAMS_KEY))) {
-          if (0U == ns_tracker->num_node_ns) {
```

Comment:
In the current unit tests, it was only testing the case of a node with a namespace. With no namespace, this error occurs when a yaml file includes both parameters (PARAMS_KEY) and descriptors (PARAM_DESCRIPTORS_KEY) under the same scope.

For example, this yaml would cause the error:
```
parameters_node:
  ros__parameters:
    param1: 30
    param3: "hello world"
  ros__parameter_descriptors:
    param1:
      type: 2
      min_value: 5
      max_value: 100
      step: 5
      read_only: false
      description: "int parameter"
      additional_constraints: "only multiples of 5"
    param2:
      min_value: 1.0
      max_value: 10.0
      description: "double parameter"
```

If these lines are removed, then this yaml becomes acceptable and the unit tests still pass. There will still be an error if trying to put a key without a node name (which is expected), but it is not caught here. I'm still trying to determine what should be the proper solution if anyone has some thoughts to add.